gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsInstrumentation.java

```diff
@@ -354,16 +341,143 @@ private void incrementMutableCounter(String name, long count) {
   }
 
   /**
-   * Counter Metrics updation based on the Http response
+   * Checking is the metric is Status Statistic or not.
    *
-   * @param statusCode of ther Http response
+   * @param name metric name
+   * @return true if metric name is status statistic else return false;
    */
-  @Override
-  public void statusMetricsUpdation(int statusCode) {
-    switch (statusCode) {
-      case 429:
-        incrementCounter(GCS_CLIENT_RATE_LIMIT_COUNT);
-        break;
+  private boolean isStatusStatistic(String name) {
+
+    for (GoogleCloudStorageStatusStatistics op : GoogleCloudStorageStatusStatistics.values()) {
+      if (name == op.getSymbol()) return true;
+    }
+    return false;
+  }
+
+  /**
+   * Getting Instance of StatisStatisticListener and intializing it once as per requirements
+   *
+   * @return Instance of StatisStatisticListener
+   */
+  public synchronized StatusStatisticListener getInstanceOfStatusStatisticListener() {
+    if (STATUS_STATISTIC_INSTANCE == null) {
```

Comment:
this is not thread safe. This will be synchronized only for a given GhfsInstrumentation object, correct? May be we can declare the instance when it is defined.